eng/versioning/pom_file_version_scanner.ps1

```diff
@@ -248,6 +255,100 @@ function Confirm-Node-Is-Part-Of-Configuration {
+            Write-Error-With-Color "$($retVal)"
+        }
+    }
+
+    # Loop through the dependencies. If any dependency is in the libHash (aka, the libraries we build)
+    # then it needs to have a version element and update tag.
+    foreach($dependencyNode in $xmlPomFile.GetElementsByTagName("dependency"))
+    {
+        $artifactId = $dependencyNode.artifactId
+        $groupId = $dependencyNode.groupId
+        # If the artifactId and groupId are both empty then check to see if this
+        # is part of a configuration entry. If so then just continue.
+        if (!$artifactId -and !$groupId)
+        {
+            $isPartOfConfig = Confirm-Node-Is-Part-Of-Configuration $dependencyNode
+            if (!$isPartOfConfig)
+            {
+                $script:FoundError = $true
+                # Because this particular case is harder to track down, print the OuterXML which is effectively the entire tag
+                Write-Error-With-Color "Error: dependency is missing version element and/or artifactId and groupId elements dependencyNode=$($dependencyNode.OuterXml)"
+            }
+            continue
+        }
+        $hashKey = "$($groupId):$($artifactId)"
+        $versionNode = $dependencyNode.GetElementsByTagName("version")[0]
+        # If this is something we build and release, it better have a version and a version tag
+        if ($libHash.ContainsKey($hashKey))
+        {
+            if (-not $versionNode)
+            {
+                $script:FoundError = $true
+                Write-Error-With-Color "Error: dependency is missing version element and tag groupId=$($groupId), artifactId=$($artifactId) should be <version></version> <!-- {x-version-update;$($groupId):$($artifactId);current|dependency|external_dependency<select one>} -->"
+            } else {
+                # verify the version tag and version are correct
+                if ($versionNode.NextSibling -and $versionNode.NextSibling.NodeType -eq "Comment")
+                {
+                    $retVal = Test-Dependency-Tag-And-Version $libHash $extDepHash $versionNode.InnerText.Trim() $versionNode.NextSibling.Value
+                    if ($retVal)
+                    {
+                        $script:FoundError = $true
+                        Write-Error-With-Color "$($retVal)"
+                    }
+                } else {
+                    $script:FoundError = $true
+                    Write-Error-With-Color "Error: dependency is missing version tag groupId=$($groupId), artifactId=$($artifactId) tag should be <!-- {x-version-update;$($groupId):$($artifactId);current|dependency|external_dependency<select one>} -->"
+                }
+            }
+        } else {
+            # else, if there's a version tag verify it, otherwise just skip it since the version should be coming
+            # from the bom
+            if ($versionNode -and $versionNode.NextSibling -and $versionNode.NextSibling.NodeType -eq "Comment")
```

Comment:
What is the conditions about the NextSibling? I'd assume if there is a version tag there must also be a comment tag.

Reply:
Good catch. This should be checking if there's a version node separately...something like. I'll update it in the next commit.
`if ($versionNode)
    if ($versionNode.NextSibling -and $versionNode.NextSibling.NodeType -eq "Comment") 
    {
        ...do verification on the tag and version
    }
    else {
        Error because there's a version with no version tag
    }
`

Reply:
This was added with the latest commit. I'm resolving the conversation.